certification/results.go

```diff
@@ -12,6 +12,10 @@ type openshiftClusterVersion = runtime.OpenshiftClusterVersion
 type Result struct {
 	check.Check
 	ElapsedTime time.Duration
+	// Err contains the error a check itself throws if it failed to run.
+	// If populated, the expectation is that this Result is in the
+	// Results{}.Errors slice.
+	Err error
```

Comment:
I'd like to propose that instead of exposing a raw struct field, we extend this via a new method and a private field.

```
type Result struct {
	check.Check
	ElapsedTime time.Duration
	err error
}

func (r Result) Error() error {
	return r.err
}

func (r *Result) WithError(err error) *Result {
	r.err = err
	return r
}
```

Reply:
ok for me, fixed